src/test/java/com/cag/cagbackendapi/controller/UserControllerTest.kt

```diff
@@ -0,0 +1,58 @@
+package com.cag.cagbackendapi.controller
+
+import com.cag.cagbackendapi.constants.DetailedErrorMessages
+import com.cag.cagbackendapi.error.exceptions.UnauthorizedException
+import com.cag.cagbackendapi.model.UserModel
+import com.cag.cagbackendapi.service.user.UserService
+import com.cag.cagbackendapi.service.validation.ValidationService
+import com.nhaarman.mockito_kotlin.*
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.junit.jupiter.MockitoExtension
+
+@ExtendWith(MockitoExtension::class)
+class UserControllerTest {
+
+    private var userService: UserService = mock()
+    private var validationService: ValidationService = mock()
+
+    @InjectMocks
+    private lateinit var userController: UserController
+
+    @Test
+    fun registerUser_validInput_returns201() {
+        val testAuthKey = "testAuthKey"
+        val testUser = UserModel(name = "John Smith", email = "johnjohn@aol.com")
+
+        doNothing().whenever(validationService).validateAuthKey(testAuthKey)
```

Comment:
Making sure I understand this file, the point of this one is to create a mock user to enter their fake information into the database?

Reply:
This is just a test to validate that registering a user calls the right functions in our logic and returns the right values. A user will not actually be created as a result of tis test because this is a unit test and it uses mocks - which are different than the "real" services. They're mock/fake services that model after the real ones. The mocks don't actually create any user because they're fake for testing purposes!